lib/board.ex

```diff
@@ -27,8 +27,22 @@ defmodule Board do
   end
 
   def select_tile(board, coordinate_pair) do
-    selected_tile = get_tile(board, coordinate_pair) |> Tile.select()
-    replace_tile(board, coordinate_pair, selected_tile)
+    tile = get_tile(board, coordinate_pair)
+
+    if Tile.is_revealed?(tile) do
+      {:error, :already_selected}
+    else
+      board =
+        reveal_tile(board, coordinate_pair)
+        |> FloodFiller.flood_fill(coordinate_pair)
```

Comment:
Is there any time you _don't_ want to flood fill when you reveal a tile? (Maybe the answer is "in all the tests that have to change" :smile:). If not, I'd consider burying this deeper and making flood fill something that just happens when you reveal a tile.

Reply:
Yes, I think you're right—I'll look into this! :)

Reply:
After looking at this, I think I actually need to keep the flood fill separate from revealing the tile in this instance.. Since the user is making their very first move, there are no bombs on the board yet. Once they make their move, I place the bombs around that first move, then I can calculate the adjacent bomb counts, and only then can I flood fill the board.